File: applications/test_scripts/Live Tracking for Shipments.js
describe('Live Tracking for Shipments', () => {
  const shipmentID = '54321';
  const userID = '09876';

  before(() => {
    cy.login(userID);
  });

  beforeEach(() => {
    cy.visit('/live-tracking');
  });

  it('should display live tracking page', () => {
    cy.get('h1').should('contain', 'Live Tracking');
  });

  it('should enter shipment ID and display tracking details', () => {
    cy.get('#trackingField').type(shipmentID);
    cy.get('#trackButton').click();
    cy.get('#trackingDetails').should('contain', shipmentID);
  });

  it('should display current location on map', () => {
    cy.get('#currentLocation').should('be.visible');
  });

  it('should simulate location change and update map in real-time', () => {
    cy.simulateLocationChange();
    cy.get('#currentLocation').should('contain', 'New Location');
  });

  it('should verify estimated delivery time', () => {
    cy.get('#estimatedDeliveryTime').should('be.visible');
  });

  it('should refresh the page and maintain tracking information', () => {
    cy.reload();
    cy.get('#trackingDetails').should('contain', shipmentID);
  });

  it('should logout and login and maintain tracking information', () => {
    cy.logout();
    cy.login(userID);
    cy.get('#trackingDetails').should('contain', shipmentID);
  });

  it('should check notification settings for live tracking updates', () => {
    cy.get('#notificationSettings').should('be.checked');
  });

  it('should handle network issue gracefully and retry update', () => {
    cy.simulateNetworkIssue();
    cy.get('#retryUpdate').should('be.visible');
  });

  it('should verify tracking history log', () => {
    cy.get('#trackingHistoryLog').should('contain', 'Location Update');
  });

  it('should not display error messages during location updates', () => {
    cy.get('#errorMessages').should('not.exist');
  });

  it('should synchronize location updates across devices', () => {
    cy.simulateDeviceChange();
    cy.get('#currentLocation').should('contain', 'New Location');
  });

  it('should verify live tracking on a mobile device', () => {
    cy.viewport('iphone-6');
    cy.get('#trackingDetails').should('contain', shipmentID);
  });

  it('should maintain tracking information after system reboot', () => {
    cy.simulateSystemReboot();
    cy.get('#trackingDetails').should('contain', shipmentID);
  });

  it('should verify the accuracy of the live tracking system', () => {
    cy.get('#trackingAccuracy').should('contain', 'Accurate');
  });
});

File: src/test/wawa/test_scripts/Live Tracking for Shipments.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.testng.Assert;

public class LiveTrackingAutomation {

    private WebDriver driver;

    public LiveTrackingAutomation() {
        System.setProperty("webdriver.chrome.driver", "path/to/chromedriver");
        driver = new ChromeDriver();
    }

    public void navigateToLiveTrackingPage() {
        driver.get("http://example.com/live-tracking");
    }

    public void enterShipmentID(String shipmentID) {
        WebElement trackingField = driver.findElement(By.id("trackingField"));
        trackingField.sendKeys(shipmentID);
    }

    public void checkCurrentLocation() {
        WebElement locationElement = driver.findElement(By.id("currentLocation"));
        String location = locationElement.getText();
        Assert.assertNotNull(location, "Current location is not displayed.");
    }

    public void simulateChangeInShipmentLocation() {
        // Simulate location change logic here
    }

    public void verifyEstimatedDeliveryTime() {
        WebElement deliveryTimeElement = driver.findElement(By.id("estimatedDeliveryTime"));
        String deliveryTime = deliveryTimeElement.getText();
        Assert.assertNotNull(deliveryTime, "Estimated delivery time is not displayed.");
    }

    public void refreshPage() {
        driver.navigate().refresh();
    }

    public void logOutAndLogBackIn() {
        WebElement logoutButton = driver.findElement(By.id("logoutButton"));
        logoutButton.click();
        WebElement loginButton = driver.findElement(By.id("loginButton"));
        loginButton.click();
    }

    public void checkNotificationSettings() {
        WebElement notificationSettings = driver.findElement(By.id("notificationSettings"));
        Assert.assertTrue(notificationSettings.isDisplayed(), "Notification settings are not visible.");
    }

    public void simulateNetworkIssueAndUpdateLocation() {
        // Simulate network issue and update location logic here
    }

    public void verifyTrackingHistoryLog() {
        WebElement historyLog = driver.findElement(By.id("trackingHistoryLog"));
        Assert.assertTrue(historyLog.isDisplayed(), "Tracking history log is not visible.");
    }

    public void checkErrorMessagesDuringLocationUpdates() {
        WebElement errorMessage = driver.findElement(By.id("errorMessage"));
        Assert.assertFalse(errorMessage.isDisplayed(), "Error message is displayed during location updates.");
    }

    public void updateLocationFromDifferentDevice() {
        // Logic to update location from a different device
    }

    public void verifyLiveTrackingOnMobileDevice() {
        // Logic to verify live tracking on a mobile device
    }

    public void checkLiveTrackingAfterSystemReboot() {
        // Logic to check live tracking after system reboot
    }

    public void verifyAccuracyOfLiveTrackingSystem() {
        // Logic to verify accuracy of the live tracking system
    }

    public void closeBrowser() {
        driver.quit();
    }

    public static void main(String[] args) {
        LiveTrackingAutomation automation = new LiveTrackingAutomation();
        automation.navigateToLiveTrackingPage();
        automation.enterShipmentID("54321");
        automation.checkCurrentLocation();
        automation.simulateChangeInShipmentLocation();
        automation.verifyEstimatedDeliveryTime();
        automation.refreshPage();
        automation.logOutAndLogBackIn();
        automation.checkNotificationSettings();
        automation.simulateNetworkIssueAndUpdateLocation();
        automation.verifyTrackingHistoryLog();
        automation.checkErrorMessagesDuringLocationUpdates();
        automation.updateLocationFromDifferentDevice();
        automation.verifyLiveTrackingOnMobileDevice();
        automation.checkLiveTrackingAfterSystemReboot();
        automation.verifyAccuracyOfLiveTrackingSystem();
        automation.closeBrowser();
    }
}